### PR TITLE
feat: defi opportunity

### LINKED ISSUE #724: Discover page
Loom : https://www.loom.com/share/f8fed2d9461043608869e7b56b0bc06b

Figma file : https://www.figma.com/design/fh0OAvj4AS08kHoSxu3DkE/%F0%9F%9A%80-Starknet-Quest?node-id=5448-5136&t=7G045GpFt0Wt8yqV-1

Ressources : 
- Component: https://ui.shadcn.com/examples/tasks
- Notion: https://www.notion.so/lfg-labs/Discover-SQ-Page-3faf27010c004466a7529b92a5f4b401

### PR DESCRIPTION
**Pull request by ayushtom**
close #724

navigate to `/discover/defi` to check changes

## Summary by CodeRabbit

- **New Features**
  - Introduced a new DeFi statistics page displaying derivatives, lending, pairing, and alternative protocol data.
  - Added a `DataTable` component for customizable filtering and sorting of DeFi-related information.
  - Added an `ActionText` component to display icons and text dynamically.
  - Added a new "Earn" link to the navbar menu.

- **Enhancements**
  - Updated the `Dropdown` component to set the initial selected value based on the `value` prop.
  - Introduced new table components with updated styling classes for creating interactive tables.

- **Constants**
  - Added predefined values for token options, stables, airdrop apps, and audited apps.

- **Utilities**
  - Added a utility function for mapping app names to specific action URLs.

- **Refactors**
  - Refactored the leaderboard page for improved readability and styling.

### CODE REVIEW

**Review by fricoben — CHANGES_REQUESTED** [review]
## E2E Feedback

https://www.loom.com/share/21a230cb51bd4cc283a4c22b6fae58d1

Checking the code now

File: app/discover/defi/page.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+} from "@services/apiService";
+import React, { useEffect, useCallback } from "react";
+import styles from "@styles/discover.module.css";
+import Typography from "@components/UI/typography/typography";
+import { TEXT_TYPE } from "@constants/typography";
+
+export default function Page() {
+  const [data, setData] = React.useState<TableInfo[]>([]);
+  const [loading, setLoading] = React.useState<boolean>(false);
+  const fetchPageData = useCallback(async () => {
```
Comment (fricoben) [inline app/discover/defi/page.tsx:16]: Create callback outside of the component :)

**Review by fricoben — CHANGES_REQUESTED** [review]
## E2E and code

Didn't check the code too much as it's individual feature it should make anything bug

File: app/discover/defi/page.tsx (hunk 1 above)
Comment (ayushtom) [reply app/discover/defi/page.tsx:16]: done

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**